internal/images/image_utils.go

```diff
@@ -151,29 +152,63 @@ func GetPrometheusImageRegistry() string {
 	return imageRegistry
 }
 
-func GetSha(imageName string) string {
-	// Pull the image
-	pullCmd := exec.Command("docker", "pull", imageName)
-	if err := pullCmd.Run(); err != nil {
-		fmt.Printf("Error pulling image: %v", err)
-		return err.Error()
-	}
+func AddShaToImages(imageList []SkupperImage) []SkupperImage {
+	var completedImages []SkupperImage
+	var waitGroup sync.WaitGroup
+	imageChannel := make(chan SkupperImage, len(imageList))
+
+	for _, image := range imageList {
+
+		waitGroup.Add(1)
+
+		go func(skImage SkupperImage) {
+			defer waitGroup.Done()
+
+			var output bytes.Buffer
+			checkCmd := exec.Command("docker", "inspect", "--format={{index .RepoDigests 0}}", skImage.Name)
```

Comment:
What do you think of running a `docker pull` before the `inspect`? 
This might be useful when using tags that are constantly overwritten like `v2-dev` or `main`.

Reply:
What about doing it only if "tag" does not start with a number?
This way, when dealing with a particular release, this won't be done all the time.

Maybe not needed at all, but the reason why it might be useful to me would be when someone
is using v2-dev (for example) as the local image version might not have the same digest of the
image actually in use in the cluster.

Reply:
I was thinking something like having an array of always-pull-this-tags = [main, v2-dev]

what do you think?

Reply:
It works for me.

Reply:
To some extent this code is trying to do two very different things. One is to determine what image to use when configuring a site. The other is to report what is actually in use. I wonder f actually we need to be clearer about those distinct goals, including within the version command itself.

Reply:
That is a good point.

These images are no longer used by the CLI to determine what a given Kubernetes site will use, right?
But of course, it could also report which images a given site (namespace) is using and the respective image
of the owning controller instance (to be evaluated).

For non-kubernetes sites, then the images in use by the CLI (from either ENV vars or the compiled ones),
might differ from what is currently active on a given namespace.

Maybe we need a separate tracker or discussion to address this issue?

Reply:
> Maybe we need a separate tracker or discussion to address this issue?
Done: #2019 
Feel free to add more comments to it.